src/base/TilesRendererBase.js

```diff
@@ -420,19 +458,20 @@ export class TilesRendererBase {
 
 		if ( isExternalTileSet ) {
 
-			downloadQueue.add( tile, tile => {
+			downloadQueue.add( tile, tileCb => {
 
 				// if it has been unloaded then the tile has been disposed
-				if ( tile.__loadIndex !== loadIndex ) {
+				if ( tileCb.__loadIndex !== loadIndex ) {
 
 					return Promise.resolve();
 
 				}
 
-				const uri = this.preprocessURL ? this.preprocessURL( tile.content.uri ) : tile.content.uri;
-				return this.fetchTileSet( uri, Object.assign( { signal }, this.fetchOptions ), tile );
+				const uri = this.preprocessURL ? this.preprocessURL( tileCb.content.uri ) : tileCb.content.uri;
+				return this.fetchTileSet( uri, Object.assign( { signal }, this.fetchOptions ), tileCb );
 
 			} )
+				.then( res => res.json() )
```

Comment:
I missed this previously -- `this.fetchTileSet` should already be returning the fetched json so this line shouldn't be needed.

Reply:
Removing, though some separation will be needed if moving forward with something like #190

I was trying to reconcile the return type difference between the `!isExternalTileset` tile fetch `Promise<Response>` and `fetchTileset(): Promise<Tileset>` by adding a generic `<T>` to PriorityQueue.add() and it seemed like `Promise<Response>` was more appropriate vs the previous `Promise<any>`.

Looking at it again I can see that is not correct, as the downloadQueue sorts `<TileInternal | Tileset>` which as a type would just make *everything* more awkward.

Reply:
Thanks! Yeah I agree -- I think it's okay to rework it in the long run I just want to keep it from breaking in the mean time.

Reply:
Totally, I'm pretty heavy on types.. let that tooling catch as many errors as I can get it to catch ;)

We'll work it out over the next couple weeks I think as change requirements come up!